server/service/integration_logger_test.go

```diff
@@ -143,17 +144,63 @@ func (s *integrationLoggerTestSuite) TestSubmitStatusLog() {
 	})
 	require.NoError(t, err)
 
+	// submit status logs
 	req := submitLogsRequest{
-		NodeKey: "1234",
+		NodeKey: h.NodeKey,
 		LogType: "status",
 		Data:    nil,
 	}
 	res := submitLogsResponse{}
 	s.DoJSON("POST", "/api/v1/osquery/log", req, http.StatusOK, &res)
 
 	logString := s.buf.String()
-	assert.Equal(t, 1, strings.Count(logString, "\"ip_addr\""))
+	assert.Equal(t, 1, strings.Count(logString, `"ip_addr"`))
+	assert.Equal(t, 1, strings.Count(logString, "x_for_ip_addr"))
+	s.buf.Reset()
+
+	// submit results logs
+	req = submitLogsRequest{
+		NodeKey: h.NodeKey,
+		LogType: "result",
+		Data:    nil,
+	}
+	res = submitLogsResponse{}
+	s.DoJSON("POST", "/api/v1/osquery/log", req, http.StatusOK, &res)
+
+	logString = s.buf.String()
+	assert.Equal(t, 1, strings.Count(logString, `"ip_addr"`))
+	assert.Equal(t, 1, strings.Count(logString, "x_for_ip_addr"))
+	s.buf.Reset()
+
+	// submit invalid type logs
+	req = submitLogsRequest{
+		NodeKey: h.NodeKey,
+		LogType: "unknown",
+		Data:    nil,
+	}
+	var errRes map[string]string
+	s.DoJSON("POST", "/api/v1/osquery/log", req, http.StatusInternalServerError, &errRes)
+	assert.Contains(t, errRes["error"], "unknown log type")
+	s.buf.Reset()
+
+	// submit gzip-encoded request
+	var body bytes.Buffer
+	gw := gzip.NewWriter(&body)
+	_, err = fmt.Fprintf(gw, `{
+		"node_key": %q,
+		"log_type": "status",
+		"data":     null
+	}`, h.NodeKey)
+	require.NoError(t, err)
+	require.NoError(t, gw.Close())
+
+	s.DoRawWithHeaders("POST", "/api/v1/osquery/log", body.Bytes(), http.StatusOK, map[string]string{"Content-Encoding": "gzip"})
+	logString = s.buf.String()
+	assert.Equal(t, 1, strings.Count(logString, `"ip_addr"`))
 	assert.Equal(t, 1, strings.Count(logString, "x_for_ip_addr"))
+
+	// submit same payload without specifying gzip encoding fails
+	s.DoRawWithHeaders("POST", "/api/v1/osquery/log", body.Bytes(), http.StatusInternalServerError, nil)
```

Comment:
Not sure if this makes sense for now, but we might want to fail with a 4xx instead of a 500 in this case.

Reply:
Agreed, see my comment elsewhere in this PR (I created an issue to do a thorough review of our status codes, we often return 500 when it should be a 4xx).